src/plots/cartesian/select.js

```diff
@@ -278,7 +278,7 @@ module.exports = function prepSelect(e, startX, startY, dragOptions, mode) {
             else {
                 // TODO: remove in v2 - this was probably never intended to work as it does,
                 // but in case anyone depends on it we don't want to break it now.
-                gd.emit('plotly_selected', undefined);
+                gd.emit('plotly_selected', {points: [], range: null});
```

Comment:
That gets triggered whenever we click on a point in select/lasso mode. The event receives `undefined` data object, which makes user check it manually. Simple compatibility is better.

Reply:
Sure, simple compatibility is better. That's why we'll do this in v2. Until then, we must preserve backward compatibility. Please undo this change.